tests/collaborative/collaborative_monkey_party.test.ts

```diff
@@ -0,0 +1,288 @@
+      const valuesUserA = getRangeValues(users[i], sheetZoneXc, sheetId);
+      const valuesUserB = getRangeValues(users[i + 1], sheetZoneXc, sheetId);
+      if (!deepEquals(valuesUserA, valuesUserB)) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+function run(network: MockTransportService, users: Model[], actions: UserAction[][]) {
+  const executed: UserAction[][] = [];
+  for (const commandGroup of actions) {
+    const concurrentlyExecuted: UserAction[] = [];
+    executed.push(concurrentlyExecuted);
+    try {
+      network.concurrent(() => {
+        for (const { command, user } of commandGroup) {
+          const result = user.dispatch(command.type, command);
+          if (result.isSuccessful) {
+            concurrentlyExecuted.push({ command, user });
+          }
+        }
+      });
+      expect(evaluationsAreSynced(users)).toBe(true);
+    } catch (e) {
+      console.error(e);
+      concurrentlyExecuted.length = 0;
+      concurrentlyExecuted.push(...commandGroup);
+      return {
+        fail: true,
+        executedActions: executed,
+      };
+    }
+  }
+  return {
+    fail: !areSynced(users),
+    executedActions: executed,
+  };
+}
+
+function randomizeCommandsPayload(commands: CoreCommand[]) {
+  const allSheetIds = commands.map((cmd) => cmd.sheetId).filter(Boolean);
+  for (const command of commands) {
+    if (isSheetDependent(command)) {
+      command.sheetId = randomChoice(allSheetIds);
+    }
+    if (isTargetDependent(command)) {
+      command.target = Array.from(range(1, randomIntFromInterval(2, 3)), generateRandomZone);
+    } else if (isPositionDependent(command)) {
+      command.col = randomIntFromInterval(0, 10);
+      command.row = randomIntFromInterval(0, 10);
+    } else if (isRangeDependant(command)) {
+      command.ranges = command.ranges.map((range) => ({
+        _sheetId: randomChoice(allSheetIds),
+        _zone: generateUnboundedRandomZone(),
+      }));
+    }
+  }
+}
+
+function generateUnboundedRandomZone(): UnboundedZone {
+  switch (randomChoice(["bounded", "full-row", "full-col"])) {
+    case "full-row": {
+      const hasHeader = Math.random() > 0.5;
+      return {
+        hasHeader,
+        left: !hasHeader ? 0 : randomIntFromInterval(0, 10),
+        right: undefined,
+        top: randomIntFromInterval(0, 10),
+        bottom: randomIntFromInterval(0, 10),
+      };
+    }
+    case "full-col": {
+      const hasHeader = Math.random() > 0.5;
+      return {
+        hasHeader,
+        left: randomIntFromInterval(0, 10),
+        right: randomIntFromInterval(0, 10),
+        top: !hasHeader ? 0 : randomIntFromInterval(0, 10),
+        bottom: undefined,
+      };
+    }
+    default:
+      return generateRandomZone();
+  }
+}
+
+function generateRandomZone() {
+  return {
+    left: randomIntFromInterval(0, 10),
+    right: randomIntFromInterval(0, 10),
+    top: randomIntFromInterval(0, 10),
+    bottom: randomIntFromInterval(0, 10),
```

Comment:
Can easily generate zones with bottom > top. Idk if we handle those or if half of the allowDispatch will fail